src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/RcwAroundCcwTests.cs

```diff
@@ -89,35 +108,48 @@ public void IInterface()
         {
             var iint = CreateWrapper<IIntImpl, IInt>();
             var obj = CreateWrapper<IInterfaceImpl, IInterface>();
-            obj.Set(iint);
+            obj.SetInt(iint);
             _ = obj.Get();
+            obj.SwapRef(ref iint);
+            obj.InInt(in iint);
+            obj.GetOut(out var _);
         }
 
         [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/88111")]
```

Comment:
I believe this was always failing in the wrong way. We are checking for an argument exception from the marshaller, but I think we end up getting one from the Span constructor in the cleanup stage. The span pointer is undefined (SkipInit'ed), and I think in x86, this leads to a null ref, but in x64 it is an argument exception. I'm working on validating this, and I'll change the tests to throw a custom exception so we can be sure we get the expected behavior.

Reply:
Okay, it looks like it was passing previously because the implementation would only return 3 elements, even though the marshal attribute said it would return 10, so it never threw in the CCW when marshalling to native, but it would throw when unmarshalling 10 elements. So this code never properly worked and we need to fix the issue where we clean up before when an array is null/undefined or a numElements is not set.

Reply:
Filed https://github.com/dotnet/runtime/issues/89747